addOns/authhelper/src/test/java/org/zaproxy/addon/authhelper/SessionDetectionScanRuleUnitTest.java

```diff
@@ -416,6 +403,17 @@ void shouldFindTokenWhenOneIsPreviouslyUnknown() throws Exception {
                         new HttpResponseHeader("HTTP/1.1 200 OK\r\n"),
                         new HttpResponseBody("<html></html>"));
 
+        history = new ArrayList<>();
+        historyProvider = new SuccessHistoryProvider(msg2);
+        AuthUtils.setHistoryProvider(historyProvider);
+
+        Control.initSingletonForTesting(model, extensionLoader);
+        Model.setSingletonForTesting(model);
+
+        Session session = mock(Session.class);
+        given(session.getContextsForUrl(anyString())).willReturn(Arrays.asList());
+        given(model.getSession()).willReturn(session);
```

Comment:
Are all these changes needed?

Reply:
It isn't actually as much change as GitHub claims. Only one line was actually changed, the block of code needed to be relocated due to order of operations.